### ISSUE TITLE
[enhancement] Support multi-user and authentication

### ISSUE THREAD

**Issue by amirvenus** [TesAnti/LangChainChat#1]
As a user, I would like to be able to have my own credentials so I can keep the history of my chats to myself. 

As an admin, I would like to define users and/or accept registarations. (possibly similar to https://github.com/open-webui/open-webui ) 
As an admin, I can choose which models are available to which users.

**Comment by TesAnti** [comment]
Good idea. But there is so many authorization and authentication techniques... Plus this would require a bunch of user management functionality which takes long time to do.
I can think of JWT approach where user is getting authorized by some external service. What do you think?
